src/components/organisms/events/table.tsx

```diff
@@ -27,14 +28,21 @@ export const EventsTable = () => {
 	const [isInitialLoad, setIsInitialLoad] = useState(true);
 
 	const [leftSideWidth] = useResize({ direction: "horizontal", initial: 50, max: 90, min: 10, id: resizeId });
-	const { eventId } = useParams();
 	const { items: sortedEvents, requestSort, sortConfig } = useSort<BaseEvent>(events || []);
+	const { connectionId, eventId, projectId, triggerId } = useParams();
 	const navigate = useNavigate();
 
+	const projectEventId = triggerId || connectionId;
+
 	useEffect(() => {
 		if (isInitialLoad) {
 			setIsInitialLoad(false);
 		}
+		if (projectEventId) {
+			fetchEvents(true, projectEventId);
+
+			return;
+		}
```

Comment:
Display loader while you are fetching the events. Because when we open connection/trigger events, we see the old events (from the last time) and only then the new loaded.

Reply:
refactored